pallets/parachain-staking/src/try_state.rs

```diff
@@ -0,0 +1,202 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::{ensure, traits::Get};
+use kilt_support::test::convert_error_message;
+use scale_info::prelude::format;
+use sp_runtime::{traits::Zero, SaturatedConversion, Saturating};
+
+use crate::{
+	set::OrderedSet,
+	types::{BalanceOf, Candidate, Stake},
+	CandidatePool, Config, DelegatorState, LastDelegation, MaxCollatorCandidateStake, MaxSelectedCandidates, Pallet,
+	Round, TopCandidates, TotalCollatorStake,
+};
+
+pub(crate) fn do_try_state<T: Config>() -> Result<(), &'static str> {
```

Comment:
Maybe I didn't see that but do we check that all `DelegatorState`s have a collator where they stake?

Reply:
It's added now